cmd/scollector/collectors/rabbitmq.go

```diff
@@ -0,0 +1,898 @@
+package collectors
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"bytes"
+
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+	"bosun.org/slog"
+)
+
+type MetricMetaFunc struct {
+	ConvertFunc func(interface{}) (float64, error)
+}
+
+const (
+	defaultRabbitmqURL string = "http://guest:guest@localhost:15672"
+)
+
+func init() {
+	collectors = append(collectors, &IntervalCollector{F: c_rabbitmq_overview, Enable: enableRabbitmq})
+	collectors = append(collectors, &IntervalCollector{F: c_rabbitmq_queues, Enable: enableRabbitmq})
+	collectors = append(collectors, &IntervalCollector{F: c_rabbitmq_nodes, Enable: enableRabbitmq})
+}
+
+// RabbitMQ registers a RabbitMQ collector.
+func RabbitMQ(url string) error {
+	safeURL, err := urlUserHost(url)
+	if err != nil {
+		return err
+	}
+	collectors = append(collectors,
+		&IntervalCollector{
+			F: func() (opentsdb.MultiDataPoint, error) {
+				return rabbitmqOverview(url)
+			},
+			name: fmt.Sprintf("rabbitmq-overview-%s", safeURL),
+		},
+		&IntervalCollector{
+			F: func() (opentsdb.MultiDataPoint, error) {
+				return rabbitmqNodes(url)
+			},
+			name: fmt.Sprintf("rabbitmq-nodes-%s", safeURL),
+		},
+		&IntervalCollector{
+			F: func() (opentsdb.MultiDataPoint, error) {
+				return rabbitmqQueues(url)
+			},
+			name: fmt.Sprintf("rabbitmq-queues-%s", safeURL),
+		})
+	return nil
+}
+
+func enableRabbitmq() bool {
+	return enableURL(defaultRabbitmqURL)()
+}
+
+func c_rabbitmq_overview() (opentsdb.MultiDataPoint, error) {
+	return rabbitmqOverview(defaultRabbitmqURL)
+
+}
+func c_rabbitmq_nodes() (opentsdb.MultiDataPoint, error) {
+	return rabbitmqNodes(defaultRabbitmqURL)
+}
+func c_rabbitmq_queues() (opentsdb.MultiDataPoint, error) {
+	return rabbitmqQueues(defaultRabbitmqURL)
+}
+
+func rabbitmqOverview(url string) (opentsdb.MultiDataPoint, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	res, err := client.Get(url + "/api/overview")
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	content, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	var overview map[string]interface{}
```

Comment:
I would much rather see a real struct type for this than map[string]interface{}. Can you paste me some of the actual json blobs in a gist or something?